### PR TITLE
[C++] [lua] Removed nested buffing logic from doRoamTick, add doBuffTick

### PR DESCRIPTION
**Pull request by Skold177**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

As the title says, this PR removes nested logic for out of combat spell casting (buffing) and moves it into its own function that is checked before roaming. This means in general, mobs will buff themselves more often, and it will be especially obvious in BCNMs (this represents a significant challenge increase for most of them)

This now means that you can control the mobs out of combat buffing frequency simply by adjusting their spell cooldowns. 

I spot checked Ro'Meave on retail and wiped out all of the RDM weapons, on spawn, they buff every 15 seconds seconds or so, so following the mobs actual spell cooldown should be fine.  

However, this PR is only half of the fix needed - it would seem most buffs are only refreshed when they fall off. Stuff like Stoneskin, Protect, Shell etc. (The exception being Bards, they cast nonstop like clockwork.) We don't have that logic built in yet, but it could be easily added soon. Other than the increased buffing frequency, I didn't notice any negative side effects. (We also could not add this functionality and just make lua spell lists instead for every mob, which will follow this logic perfectly)

I checked...
* Mobs in BCNMs, like the bards in Prehistoric Pigeons, and RDM Maat
* Weapons in Ro'Meave
* Sahagins in SSG
* Yagudos at the top of Oztroja
* Summoner mobs, to ensure they were still summoning their spirits properly (even though thats wrong)
* Also added back a missing function to calculate spell cooldowns for Elementals, and made sure Light Spirit works okay. 

## Steps to test these changes

Check all of the stuff listed above, and any other casting edge cases you can think of.

### CODE REVIEW

File: scripts/globals/summon.lua [hunk 1]
```diff
@@ -13,6 +13,14 @@ local trialSizeBattles = set{
     xi.battlefield.id.TRIAL_SIZE_TRIAL_BY_WATER,
 }
 
+xi.summon.getSummoningSkillOverCap = function(avatar)
+    local summoner       = avatar:getMaster()
+    local summoningSkill = summoner:getSkillLevel(xi.skill.SUMMONING_MAGIC)
+    local maxSkill       = summoner:getMaxSkillLevel(avatar:getMainLvl(), xi.job.SMN, xi.skill.SUMMONING_MAGIC)
+
+    return math.max(summoningSkill - maxSkill, 0)
+end
```
Comment (Xaver-DaRed) [inline scripts/globals/summon.lua:23]: 2 things:
- Why is this function global? Where is it used?
- max skill is using avatar level, which isnt necesarily the summoner level. If its anything like pets, it will be higher with decent gear. Max player level is lvl 99. Max avatar lvl is 123

**Review by Xaver-DaRed — APPROVED** [review]
